src/tls13handshake.rs

```diff
@@ -298,9 +298,20 @@ fn put_server_signature(
         let cert = parse_server_certificate(&algs, sc)?;
         let tx = transcript_add1(tx, sc);
         let th_sc = get_transcript_hash(&tx)?;
-        let pk = verification_key_from_cert(&cert)?;
+        let (signature_scheme, certificate_key) = verification_key_from_cert(&cert)?;
         let sig = parse_certificate_verify(&algs, scv)?;
         let sigval = PREFIX_SERVER_SIGNATURE.concat(&th_sc);
+        // TODO(Refactor): This is used because the API of `evercrypt_cryptolib::verify` wants a `PublicVerificationKey`.
+        //                 That enum is directly unpacked in `verify`, though.
+        let pk = match signature_scheme {
```

Comment:
This (`let match`) is not allowed in hacspec.

Reply:
Right. Didn't verified it yet because priority was to make it build again :-) But let's not introduce code that is invalid according to hacpsec.